addOns/exim/src/main/java/org/zaproxy/addon/exim/ImportExportApi.java

```diff
@@ -54,17 +58,28 @@ public String getPrefix() {
     public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
         LOG.debug("handleApiAction {} {}", name, params);
 
+        boolean success;
+        File file;
         switch (name) {
             case ACTION_IMPORTHAR:
-                File file = new File(ApiUtils.getNonEmptyStringParam(params, PARAM_FILE_PATH));
-                boolean success = HarImporter.importHarFile(file);
-                if (success) {
-                    return ApiResponseElement.OK;
-                } else {
-                    throw new ApiException(Type.BAD_EXTERNAL_DATA, file.getAbsolutePath());
-                }
+                file = new File(ApiUtils.getNonEmptyStringParam(params, PARAM_FILE_PATH));
+                success = HarImporter.importHarFile(file);
+                return handleFileImportResponse(success, file);
+            case ACTION_IMPORTURLS:
+                file = new File(ApiUtils.getNonEmptyStringParam(params, PARAM_FILE_PATH));
+                success = UrlsImporter.importUrlFile(file);
+                return handleFileImportResponse(success, file);
             default:
                 throw new ApiException(Type.BAD_ACTION);
         }
     }
+
+    private ApiResponseElement handleFileImportResponse(boolean success, File file)
+            throws ApiException {
+        if (success) {
+            return ApiResponseElement.OK;
+        } else {
```

Comment:
The else could be removed.

Reply:
Won't that mean throwing bad action when the import is unsuccessful?

Reply:
I mean just remove the else statement, it would still throw the `BAD_EXTERNAL_DATA` exception.